### PR TITLE
Switch to ImmutableMetadata

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

- Upgrades OpenMLS to the latest version, which includes a simplified version of Immutable Metadata
- Upgrades a few dependencies to the latest version, which will allow us to do the next upgrade of OpenMLS

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -30,10 +30,10 @@ futures = "0.3.30"
 futures-core = "0.3.30"
 hex = "0.4.3"
 log = "0.4.20"
-openmls = { git = "https://github.com/xmtp/openmls", branch = "main" }
-openmls_basic_credential = { git = "https://github.com/xmtp/openmls", branch = "main" }
-openmls_rust_crypto = { git = "https://github.com/xmtp/openmls", branch = "main" }
-openmls_traits = { git = "https://github.com/xmtp/openmls", branch = "main" }
+openmls = { git = "https://github.com/xmtp/openmls", rev = "0da7dcb" }
```
Comment (neekolas) [inline Cargo.toml:33]: I don't know about the rest of you, but I've been having weird issues with Rust Analyzer caching dependencies very aggressively. Even after running cargo update, it keeps the old version for type checking (but builds work fine when my code uses newer types). This persists even after running `cargo clean`. Pinning to a specific rev breaks the cache.

Comment (neekolas) [reply Cargo.toml:33]: Reloading the workspace and restarting the server also didn't help

File: xmtp_mls/src/groups/mod.rs [hunk 2]
```diff
@@ -364,15 +364,9 @@ fn build_protected_metadata_extension(
...
-    let protected_metadata = ProtectedMetadata::new(
-        &identity.installation_keys,
-        identity.application_id(),
-        identity.credential()?,
-        identity.installation_keys.to_public_vec(),
-        metadata.try_into()?,
-    )?;
-
-    Ok(Extension::ProtectedMetadata(protected_metadata))
+    let protected_metadata = Metadata::new(metadata.try_into()?);
```
Comment (tuddman) [inline xmtp_mls/src/groups/mod.rs:367]: nit about naming:   If its no longer instantiating a `ProtectedMetadata`, as that's being replaced with an `ImmutableMetadata`.

```suggestion
    let immutable_metadata = Metadata::new(metadata.try_into()?);
```

**Review by tuddman — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (neekolas) [reply Cargo.toml:33]: Woah. Maybe

**Pull request merged**